### PR TITLE
Adds a tool to tag blocks within a schematic

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
Adds the ability to tag blocks in a schematic with strings.
The data is saved in anchour TE's which implement the IBlueprintDataProvider interface.
For usage you have a new additional tool, called tagtool.
First you select a fitting anchor with the tag tool (in structurize thats only the placeholder block)
and then you rightclick in air to set a tag for the tool to use, afterwards you can mark/unmark blocks with leftclick on them.
The selected anchor and tagged blocks are rendered and highlighted.
image

Anchors also save schematic name & corners, so they can aswell identify the schematic they are in and its size.

Review please

### CODE REVIEW

File: src/main/java/com/ldtteam/structurize/blocks/PlaceholderBlock.java [hunk 1]
```diff
@@ -45,7 +44,7 @@ public ActionResultType onBlockActivated(
     {
         if (worldIn.isRemote)
         {
-            new WindowPlaceholderblock(pos).open();
+            // new WindowPlaceholderblock(pos).open();
         }
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/structurize/blocks/PlaceholderBlock.java]: Why?

File: src/main/java/com/ldtteam/structurize/items/ItemTagTool.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    @Override
+    public AbstractItemWithPosSelector getRegisteredItemInstance()
+    {
+        return ModItems.tagTool;
+    }
+
+    @Override
+    public ActionResultType onAirRightClick(final BlockPos start, final BlockPos end, final World worldIn, final PlayerEntity playerIn, final ItemStack itemStack)
+    {
+        if (!worldIn.isRemote)
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/structurize/items/ItemTagTool.java]: Idon't like empty ifs

File: src/main/java/com/ldtteam/structurize/placementhandlers/PlacementHandlers.java [hunk 3]
```diff
@@ -734,49 +737,105 @@ public Object handle(
...
+          @NotNull final BlockState blockState,
+          @Nullable final CompoundNBT tileEntityData,
+          final boolean complete,
+          final BlockPos centerPos)
+        {
+            if (tileEntityData != null)
+            {
+                if (tileEntityData.contains(TAG_CONTAINED_BLOCK))
+                {
+                    ItemStack contained = ItemStack.read(tileEntityData.getCompound(TAG_CONTAINED_BLOCK));
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/structurize/placementhandlers/PlacementHandlers.java]: do we want this stuff still?

File: src/main/java/com/ldtteam/structurize/blocks/PlaceholderBlock.java (hunk 1 above)
Comment (someaddons) [reply src/main/java/com/ldtteam/structurize/blocks/PlaceholderBlock.java]: its broken and placeholder functionality isnt used anymore

File: src/main/java/com/ldtteam/structurize/items/ItemTagTool.java (hunk 2 above)
Comment (someaddons) [reply src/main/java/com/ldtteam/structurize/items/ItemTagTool.java]: oh ye thought I needed this but didnt

File: src/main/java/com/ldtteam/structurize/placementhandlers/PlacementHandlers.java (hunk 3 above)
Comment (someaddons) [reply src/main/java/com/ldtteam/structurize/placementhandlers/PlacementHandlers.java]: nah no need

File: src/main/java/com/ldtteam/structurize/blocks/PlaceholderBlock.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/ldtteam/structurize/blocks/PlaceholderBlock.java]: Then just remove this if completely

**Comment by Raycoms** [comment]
:shipit:

File: src/main/java/com/ldtteam/structurize/client/gui/WindowTagTool.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+            /**
+             * Inserts the elements into each row.
+             * @param index the index of the row/list element.
+             * @param rowPane the parent Pane for the row, containing the elements to update.
+             */
+            @Override
+            public void updateElement(final int index, @NotNull final Pane rowPane)
+            {
+                TileEntity te = world.getTileEntity(anchorPos);
```
Comment (Nightenom) [inline src/main/java/com/ldtteam/structurize/client/gui/WindowTagTool.java:184]: ... ? seriously triggering update each tick

**Pull request merged**